### ISSUE TITLE
Keyboard accessibility: Use ENTER to select/confirm/use elements rather than only using space.

### ISSUE THREAD

**Issue by Jonas-Sander** [flutter/flutter#71998]
## Use case

When running Flutter apps in the web or on desktop developers want to have accessible apps by default.  
This includes creating apps that are usable only by keyboard.  
In this regard Flutter has built in tab-navigation and interactivity with elements (e.g. to press a button in focus) by pressing "Space".

What I as a normal web and desktop user found very surprising is that elements like buttons etc. can not be interacted with by using the "Enter"-Key by default. This is very different from what I would expect from a website or desktop app and might catch users off guard that rely on normal keyboard interaction semantics. 

If you go e.g. to the [Flutter Gallery Web-App](https://gallery.flutter.dev/ ) you can try it out yourself.  
It is possible to navigate with "TAB" and one can interact by pressing "Space". Nothing happens when pressing "Enter".

For example the Accessible Rich Internet Applications ("ARIA") standard shows that buttons should be activatable by pressing "Enter" or "Space":
https://www.w3.org/TR/wai-aria-practices-1.1/#keyboard-interaction-3

Even going on the flutter.dev website and using "TAB" to navigate you can "press" the links in the navigation on the top with the "enter"-key.  
Actually pressing space will scroll down the website and can't be used to press the links.  
https://webaim.org/techniques/keyboard/

Of course there will be differences between a web app and a website as shown above.      
Still I think using "Enter" to interact with elements so universal that Flutter should follow this behavior.  

## Proposal
One should be able to interact with elements by pressing the "Enter"-Key by default to follow common conventions in the web and desktop. This can be the same behavior as using the "Space"-Key.

**Comment by HansMuller** [comment]
CC @gspencergoog

**Comment by gspencergoog** [comment]
This should be fixed now in the master branch.
